kvmagent/kvmagent/plugins/vms/vm_host_file.py

```
@@ -0,0 +1,195 @@
+        os.makedirs(target_dir)
+    except OSError as e:
+        if e.errno != errno.EEXIST:  # ignore folder already exists error
+            raise
+
+    try:
+        with open(path, 'wb') as f:
+            f.write(raw_data)
+            f.flush()
+            os.fsync(f.fileno())
+    except (IOError, OSError) as e:
+        raise IOError("Failed to write file at %s: %s" % (path, str(e)))
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)